lib/porro/types/default.rb

```diff
@@ -13,11 +13,11 @@ def initialize(wrapper, default = nil)
       end
 
       def load(value)
-        @wrapper.load(value) || default_proc.call
+        @wrapper.load(value) || @wrapper.load(default_proc.call)
       end
```

Comment:
this line is a bit controversial...

Doing a short-ciruit here like previously `@wrapper.load(value) || default_proc.call` allows invalid values to be assigned i.e 

`attribute :gender, Porro::Types::Enum.new(%w{male female}).default('hybrid')`

will lead to `Model.new.gender` to return `'hybrid'` although it's not a valid value for the enum.

The variant `@wrapper.load(value || default_proc.call)` will fail some tests, i.e 

      attribute :gender, Porro::Types::Enum.new(%w{male female}).default('female')

      it 'converts anything else to default (female)' do
        expect(subject.send(method, 'FOOBAR')).to eq 'female'
      end

Will fail because `value == 'FOOBAR'` and `default_proc.call == 'female'` so the call looks like `@wrapper.load('FOOBAR' || 'female')` which will favor the 'FOOBAR' and the load returns `nil`.